src/app/leverage/components/your-tokens.tsx

```diff
@@ -1,25 +1,34 @@
-import clsx from 'clsx'
+import { Tab, TabGroup, TabList, TabPanel, TabPanels } from '@headlessui/react'
+import ExternalLinkIcon from '@heroicons/react/24/outline/ArrowTopRightOnSquareIcon'
+import capitalize from 'lodash/capitalize'
 import Image from 'next/image'
-import { useEffect, useState } from 'react'
+import { FC, useEffect, useMemo, useState } from 'react'
+import { Address } from 'viem'
+import { useAccount } from 'wagmi'
 
-import { useNetwork } from '@/lib/hooks/use-network'
+import { useTokenHistory } from '@/lib/hooks/use-token-history'
+import { shortenAddress } from '@/lib/utils'
+import { SkeletonLoader } from '@/lib/utils/skeleton-loader'
+import { cn } from '@/lib/utils/tailwind'
+import { mapChainIdToAddressProp } from '@/lib/utils/tokens'
 
 import { useLeverageToken } from '../provider'
 import { EnrichedToken, LeverageType } from '../types'
 import { fetchLeverageTokenPrices } from '../utils/fetch-leverage-token-prices'
 import { getLeverageBaseToken } from '../utils/get-leverage-base-token'
-import { getLeverageType } from '../utils/get-leverage-type'
+import { getLeverageAction, getLeverageType } from '../utils/get-leverage-type'
 
 const leverageTypeLabels = {
   [LeverageType.Long2x]: '2x LONG',
   [LeverageType.Long3x]: '3x LONG',
   [LeverageType.Short]: '1x SHORT',
 }
 
-export function YourTokens() {
-  const { chainId } = useNetwork()
+export const YourTokens: FC = () => {
```

Comment:
curious since I haven't seen it used much lately - favor this syntax for a specific reason?

came across this PR through some googling: https://github.com/facebook/create-react-app/pull/8177

Reply:
The reason I like this more, is its a generic type so if you do props you can do 

const Component: FC<Props> = ({ aProp, bProp }) => { ... } and im just in favor of this, vs adding type at the arguments. In general just stuck with it, nothing extraordinary or something I really adamant about. 

The children thing is another one I kinda liked, because even if children is implicitly added, you decide when to render it. But it's actually not true anymore, FC doesn't add children implicitly.

The unresolved generic  type for sure is a thing. I had need for it and then I dropped the FC for sure.